keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers import preprocessing as cv_preprocessing
+from keras_cv.utils import preprocessing as preprocessing_utils
+
+
+class RandAugment(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """RandAugment performs the Rand Augment operation on input images.
+
+    This layer can be thought of as an all in one image augmentation layer.  The policy
+    implemented by this layer has been benchmarked extensively and is effective on a
+    wide variety of datasets.
+
+    The policy operates as follows:
+
+    For each `layer`, the policy selects a random operation from a list of operations.
+    It then samples a random number and if that number is less than
+    `probability_to_apply` applies it to the given image.
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        num_layers: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.
+        magnitude_standard_deviation: the standard deviation to use when drawing values
+            for the perturbations.
+        probability_to_apply:  the probability to apply an augmentation at each layer.
+
+    Usage:
+    ```python
+    (x_test, y_test), _ = tf.keras.datasets.cifar10.load_data()
+    rand_augment = keras_cv.layers.RandAugment(
+        value_range=(0, 255), num_layers=3, magnitude=5.0
+    )
+    x_test = rand_augment(x_test)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        num_layers=3,
+        magnitude=5.0,
+        magnitude_standard_deviation=1.5,
+        probability_to_apply=None,
+        seed=None,
+        **kwargs,
```

Comment:
Should consider making the set of applicable transformations configurable?

Reply:
I was thinking about this, I wonder if we should offer a higher level abstraction:

provide us a list of layers, we create a random policy from it.  This is more or less what AutoAugment is.  Seems kinda overkill to make RandAugment too configurable.

Reply:
Overkill is the wrong word - but it seems like the "wrong abstraction" for what we want to offer.

We want to offer a pipeline that allows users to randomly apply layers from a selection of layers.  RandAugment happens to be one of those pipelines, that happens to be well researched.  But I don't think the RandAugment class should handle this responsibility